src/Avalonia.Controls/TreeViewItem.cs

```diff
@@ -166,30 +168,95 @@ protected override void OnKeyDown(KeyEventArgs e)
         {
             if (!e.Handled)
             {
-                switch (e.Key)
+                Func<TreeViewItem, bool>? handler =
+                    e.Key switch
+                    {
+                        Key.Left => ApplyRecursivelyIfCtrl(FocusAwareCollapseItem, e.KeyModifiers),
+                        Key.Right => ApplyRecursivelyIfCtrl(ExpandItem, e.KeyModifiers),
+                        Key.Enter or Key.Space => ApplyRecursivelyIfCtrl(IsExpanded ? CollapseItem : ExpandItem, e.KeyModifiers),
+
+                        // do not handle CTRL with numpad keys
+                        Key.Subtract => FocusAwareCollapseItem,
+                        Key.Add => ExpandItem,
+                        Key.Divide => ApplyToSubtree(CollapseItem),
+                        Key.Multiply => ApplyToSubtree(ExpandItem),
+                        _ => null,
+                    };
+
+                if(handler is not null)
+                {
+                    e.Handled = handler(this);
+                }
+
+
```

Comment:
nit: Extra blank line here.

Feel free to ignore some of these nits. Avalonia tries to maintain high code formatting quality but it can get monotonous and it's better to just accept new functionality rather than make it perfect.